v2/coordinator-specification.md

```diff
@@ -0,0 +1,691 @@
+    "verifyingContractAddress": "0x48bacb9266a570d521063ef5dd96e61686dbe788",
+    "signature": "0x1b36fca95c03d4e28b0ffb7796194841612005f0913e60f9da36f477a432e478240af81e8be55368b1a43f21266c6964e2b3d6c55b9d11dd5318ab76e63ff0c6f903"
+  },
+  "txOrigin": "0xe834ec434daba538cd1b9fe1582052b880bd7e63"
+}
+```
+
+- `signedTransaction` - A signed [0x transaction](https://github.com/0xProject/0x-protocol-specification/blob/ad13141d9a2c6d93e06658d18c53e9f3d99442d4/v2/v2-specification.md#transactions) along with the `verifyingContractAddress` which is part of it's [EIP712 domain](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#definition-of-domainseparator).
+- `txOrigin` - The address that will eventually submit the Ethereum transaction executing this 0x transaction on-chain. This will be enforced by the `Coordinator` extension contract.
+
+#### Response
+
+**Fill request response:**
+
+[See fill response schema](TODO)
+
+```json
+{
+  "signatures": [
+    "0x1cc07d7ae39679690a91418d46491520f058e4fb14debdf2e98f2376b3970de8512ace44af0be6d1c65617f7aae8c2364ff63f241515ee1559c3eeecb0f671d9e903"
+  ],
+  "expirationsTimeSeconds": [1552390014]
+}
+```
+
+- `signatures` - the coordinator signatures required to submit the 0x transaction
+- `expirationsTimeSeconds` - when the signatures will expire and no longer be valid
+
+Usually a single signature will be returned. Only when someone requests to batchFill multiple orders from the coordinator that were created with different supported feeRecipientAddresses, will this return a signature per feeRecipientAddress involved.
+
+**Cancellation request response:**
+
+[See cancellation response schema](TODO)
+
+```json
+{
+  "outstandingSignatures": [
+    {
+      "orderHash": "0xd1dc61f3e7e5f41d72beae7863487beea108971de678ca00d903756f842ef3ce",
+      "coordinatorSignature": "0x1c7383ca8ebd6de8b5b20b1c2d49bea166df7dfe4af1932c9c52ec07334e859cf2176901da35f4480ceb3ab63d8d0339d851c31929c40d88752689b9a8a535671303",
+      "expirationTimeSeconds": 1552390380,
+      "takerAssetFillAmount": "100000000000000000000"
+    }
+  ]
+}
+```
+
+- `outstandingSignatures` - Information about the outstanding, unexpired signatures to fill the order(s) that have been soft-cancelled. Once they expire, the maker will have 100% certainty that their order can no longer be filled.
+
+## WebSocket API
+
+### Transaction request notifications
+
+#### Endpoint: `/v1/requests`
+
+The WebSocket endpoint allows a client to subscribe to the following events:
+
+#### FILL_REQUEST_RECEIVED
+
+A fill request was received and is valid. The request has not yet been granted a coordinator signature. Depending on the server implementation, the request might need to wait for a selective delay before being issued a signature. (TODO: this should probably only broadcast the transactionHash)
```

Comment:
A note on this TODO: I think the `FILL_REQUEST_RECEIVED` message should literally just be the `transactionHash`. Think of the request-approval messages as a commit-reveal process that is mostly intended for auditability. 

If you broadcast the entire transaction, the server opens itself up to maker griefing attacks (a small order can be soft cancelled to invalidate the whole transaction). If only the hash is broadcasted, then makers will only soft cancel orders that they truly want to cancel.